experimenter/modify_generator.py

```diff
@@ -0,0 +1,162 @@
+        else:
+            raise ValueError("This type of modification is not yet an option")
+    
+ 
+    def _check_for_duplicates(self, pipeline_to_check, problem_ref_to_check):
+        """Pseudo function/method for duplicate checking 
+            - This function is not complete and will be used for future generation type jobs
+        """
+        #create the pipeline to check for duplicates from the path
+        pipeline_object = d3m.metadata.pipeline.Pipeline.from_json(pipeline_to_check)
+        #query through the database for equal pipelines
+        similar_pipeline_runs_in_database = query.generate_similar_pipeline_runs()
+        for pipeline in similar_pipeline_runs_in_database:
+            if (pipeline_object.equals(pipeline)):
+                return True
+        return False   
+    
+    
+    def _modify_random_seed(self, seed_limit, query_args):
+        """Generates new seeds for a given pipeline, problem, and dataset
+           It is dependent on the seed limit for how many it will generate
+        """
+        used_seeds = query_args['tested_seeds']
+        num_run = len(used_seeds)
+        #run until the right number of seeds have been run
+        while (num_run < seed_limit):
+            new_seed = randint(1,100000)
+            if (new_seed in used_seeds):
+                continue
+            num_run += 1
+            used_seeds.add(new_seed)
+            #yield the necessary job requirements
+            yield (query_args['pipeline'], query_args['problem_path'], query_args['dataset_doc_path'], new_seed, 
+                  (query_args['data_prep_pipeline'], query_args['data_prep_seed'], query_args['data_params']), 
+                  (query_args['scoring_pipeline'], query_args['scoring_seed'], query_args['scoring_params'])) 
+            
+            
+    def _run_seed_test(self,args):
+        """ Test designed for development and functionality purposes.
+            It uses and dataset and pipeline that is saved in d3m-experimenter
+        """
+        with open('experimenter/pipelines/bagging_classification.json', 'r') as pipeline_file:
+            pipeline = json.load(pipeline_file) 
+        dataset_path = utils.get_dataset_doc_path('185_baseball_MIN_METADATA_dataset')
+        problem_path = utils.get_problem_path('185_baseball_MIN_METADATA_problem')
+        data_prep_seed = 0
+        data_prep_seed = 0
+        data_prep_pipeline = data_split_file
+        scoring_pipeline = scoring_file
+        scoring_seed = 0
+        used_seeds = {2,15}
+        yield {'pipeline': pipeline, 'problem_path': problem_path, 'dataset_doc_path': dataset_path, 
+               'tested_seeds': used_seeds, 'data_prep_pipeline': data_prep_pipeline, 
+               'data_prep_seed': data_prep_seed, 'data_params': None,
+               'scoring_pipeline': scoring_pipeline, 'scoring_seed': scoring_seed,
+               'scoring_params': None}
```

Comment:
Let's move this to `tests/test_modify_generator.py`. You can use `tests/test_worker.py` as a template. From the root of the repo, you can run `run_tests.py` and it should automatically find the new test file you make. You need to make a class the inhereits from `unittest.TestCase` and prefix the name of all test functions with the word `test`, e.g. `test_run_seed`.

Reply:
Looks like you have already started this. Finish it up by removing this code.